tests/conftest.py

```diff
@@ -159,6 +159,9 @@ def notebooks():
         "dkn_quickstart": os.path.join(
             folder_notebooks, "00_quick_start", "dkn_synthetic.ipynb"
         ),
+        "lightgbm_quickstart": os.path.join(
```

Comment:
apart from the smoke test that you added, we should add a unit test here https://github.com/Microsoft/Recommenders/blob/master/tests/unit/test_notebooks_python.py, this just runs the notebook with as fast as possible. 

In other examples you will see that we are just runing the notebooks, but for algos that take too long, we restrict the number of epochs like in this one https://github.com/Microsoft/Recommenders/blob/master/tests/unit/test_notebooks_gpu.py#L38. This is because we don't want the unit tests to take too long

Reply:
I added a unit test for lightgbm, and it is passed. 
However, when I test the whole `tests/unit/test_notebooks_python.py`, it meets failure on the function `test_vw_deep_dive_runs`, both in my branch and master, and the `02_model/vowpal_wabbit_deep_dive.ipynb` also raises exception in Cell "In [7]". I am so confused about that...

Reply:
mmm that's weird, can you please paste the error? also, are you on windows or linux?

Reply:
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-7-b44d564263fc> in <module>
     17                 test_params=test_params,
     18                 test_data=test,
---> 19                 prediction_path=prediction_path)
     20 
     21 comparison = pd.DataFrame(result, index=['Linear Regression'])

<ipython-input-3-3fbf83133880> in run_vw(train_params, test_params, test_data, prediction_path, logistic)
     13     # train model
     14     train_start = process_time()
---> 15     run(train_params.split(' '), check=True)
     16     train_stop = process_time()
     17 

~/.conda/envs/reco_base/lib/python3.6/subprocess.py in run(input, timeout, check, *popenargs, **kwargs)
    421         kwargs['stdin'] = PIPE
    422 
--> 423     with Popen(*popenargs, **kwargs) as process:
    424         try:
    425             stdout, stderr = process.communicate(input, timeout=timeout)

~/.conda/envs/reco_base/lib/python3.6/subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors)
    727                                 c2pread, c2pwrite,
    728                                 errread, errwrite,
--> 729                                 restore_signals, start_new_session)
    730         except:
    731             # Cleanup if the child failed starting.

~/.conda/envs/reco_base/lib/python3.6/subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, start_new_session)
   1362                         if errno_num == errno.ENOENT:
   1363                             err_msg += ': ' + repr(err_filename)
-> 1364                     raise child_exception_type(errno_num, err_msg, err_filename)
   1365                 raise child_exception_type(err_msg)
   1366 

FileNotFoundError: [Errno 2] No such file or directory: 'vw': 'vw'
```
These are found when running the `02_model/vowpal_wabbit_deep_dive.ipynb` (similarly when unit testing) and I am on Linux.

Reply:
yeah, we need to skip these tests if vowpal wabbit is not installed, it's trying to run vw from subprocess and failing

Reply:
OK, now the unit testing of my commit is not successful, can I re-test it and how?